master/buildbot/status/build.py

```diff
@@ -629,6 +629,14 @@ def get_artifacts(self):
                 self.artifacts = artifacts
             return self.artifacts
 
+    def get_trigger_steps_time(self):
+        trigger = 0
+        trigger_steps = [step.finished - step.started for step in self.steps if step.finished and str(TriggerType) == step.step_type ]
```

Comment:
@kkateq 
I tried out and it doesn't work with a subclass of the trigger step using 
TriggerType = "<class 'buildbot.steps.trigger.Trigger'>"|

from buildbot.steps.trigger import Trigger

class PartitionTrigger(Trigger):
    name = "PartitionTrigger"

maybe @campos-ddc or @sualitu has a suggestion for this?